### ISSUE TITLE
mod / div for Int can introduce values that are not intended as inhabitants of Int

### ISSUE THREAD

**Issue by garyb** [purescript/purescript-prelude#169]
In 0.12 too:

```
mod 0 0 = NaN
1 / 0 = Infinity -- (although this one is not so in 0.11 - it's 0)
```

I guess these should probably both just be 0.

**Comment by hdgarrood** [comment]
Previous discussion in https://github.com/purescript/purescript-prelude/pull/103 about what division by zero should do. I'd still rather they both throw errors (this is what GHC does too, incidentally) but at the least they should both do the same thing, I think (i.e. both return 0, or both throw errors, or similar).

**Comment by garyb** [comment]
Resolved in 3c7f84ae881af003d27cf94e1477431fe4328ba0
